test/test.js

```diff
@@ -14,290 +19,476 @@ function log () {
   console.log.apply(console, args)
 }
 
-var testfile = __dirname + '/testfile.txt'
+var testfilePath = __dirname + '/testfile.txt'
+var testfile = fs.readFileSync(__dirname + '/testfile.txt')
+
+describe('IPFS Node.js API wrapper tests', function () {
+  var apiClients = {} // a, b, c
+  var ipfsNodes = {} // a, b, c
 
-describe('ipfs node api', function () {
-  var ipfs, ipfsNode
   before(function (done) {
     this.timeout(20000)
     log('ipfs node setup')
+    var counter = 0
+
+    if (isNode) {
+      startIndependentNode(ipfsNodes, apiClients, 'a', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'b', finish)
+      startIndependentNode(ipfsNodes, apiClients, 'c', finish)
```

Comment:
My thinking would be along these lines: 

1. Gulp starts x disposable nodes
2. Trigger node test run (only tests relying on nodes running)
3. Trigger browser test run
4. Gulp stops all nodes
5. Trigger node tests that involve manual start and stop of nodes (like https://github.com/ipfs/node-ipfs-api/blob/master/test/test.js#L290-L301)